include/boost/next_prior.hpp

```diff
@@ -140,6 +138,24 @@ struct prior_impl1< T, Distance, true >
     }
 };
 
+template< typename T, typename Distance, bool IsIterator = is_iterator< T >::value >
+struct prior_advance_impl :
+    public prior_minus_assign_impl< T, Distance >
+{
+};
+
+template< typename T, typename Distance >
+struct prior_advance_impl< T, Distance, true >
+{
+    static T call(T x, Distance n)
+    {
+        // Avoid negating n to sidestep possible integer overflow
+        std::reverse_iterator< T > rx(x);
```

Comment:
> I don't want to trade integer overflow for additional functionality.

I thought that the type of the second parameter `n` of `boost::next/prior` is iterator's `differece_type` (as in `std::next/prev`) and so integer overflow rarely happens. Or do you have the following situation in mind?
- The type of `n` is an unsigned integer type; or
- `sizeof(n) > sizeof(differece_type)`.

> In practice, I think std::reverse_iterator should work in the case you presented, unless it actively tests for the underlying iterator category.

Right. 

> Also, since boost::reverse_iterator already depends on boost::prior, we can't use it now. See boostorg/iterator#26.

I see. Thanks for making PR to Boost.Iterator.